crates/telio-nurse/src/aggregator.rs

```diff
@@ -76,94 +65,107 @@ impl From<EndpointProvider> for EndpointType {
-        reason: RelayConnectionChangeReason,
-    },
-    Peer {
-        endpoints: PeerEndpointTypes,
-        rx_bytes: u64,
-        tx_bytes: u64,
-    },
+pub(crate) struct PeerConnectionData {
+    endpoints: PeerEndpointTypes,
+    rx_bytes: u64,
+    tx_bytes: u64,
+}
+
+impl Serialize for PeerConnectionData {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let PeerConnectionData {
+            endpoints,
+            rx_bytes,
+            tx_bytes,
+        } = self;
+        let initiator_repr = endpoints.initiator_ep as u16;
+        let responder_repr = endpoints.responder_ep as u16;
+        let strategy_id = (initiator_repr << ENDPOINT_BIT_FIELD_WIDTH) + responder_repr;
+        let mut seq = serializer.serialize_tuple(PEER_TUPLE_LEN)?;
+        seq.serialize_element(&strategy_id)?;
+        seq.serialize_element(&rx_bytes)?;
+        seq.serialize_element(&tx_bytes)?;
+        seq.end()
+    }
+}
+
+#[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
+pub(crate) struct RelayConnectionData {
+    state: RelayConnectionState,
+    reason: RelayConnectionChangeReason,
 }
 
 const ENDPOINT_BIT_FIELD_WIDTH: u16 = 4;
 const RELAY_TUPLE_LEN: usize = 2;
 const PEER_TUPLE_LEN: usize = 3;
 
-impl Serialize for ConnectionData {
+impl Serialize for RelayConnectionData {
     fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
     where
         S: serde::Serializer,
     {
-        match self {
-            ConnectionData::Relay { state, reason } => {
-                let mut seq = serializer.serialize_tuple(RELAY_TUPLE_LEN)?;
-                seq.serialize_element(&(*state as u64))?;
-                seq.serialize_element(&(*reason as u64))?;
-                seq.end()
-            }
-            ConnectionData::Peer {
-                endpoints,
-                rx_bytes,
-                tx_bytes,
-            } => {
-                let initiator_repr = endpoints.initiator_ep as u16;
-                let responder_repr = endpoints.responder_ep as u16;
-                let strategy_id = (initiator_repr << ENDPOINT_BIT_FIELD_WIDTH) + responder_repr;
-                let mut seq = serializer.serialize_tuple(PEER_TUPLE_LEN)?;
-                seq.serialize_element(&strategy_id)?;
-                seq.serialize_element(&rx_bytes)?;
-                seq.serialize_element(&tx_bytes)?;
-                seq.end()
-            }
-        }
+        let RelayConnectionData { state, reason } = self;
+        let mut seq = serializer.serialize_tuple(RELAY_TUPLE_LEN)?;
+        seq.serialize_element(&(*state as u64))?;
+        seq.serialize_element(&(*reason as u64))?;
+        seq.end()
     }
 }
 
 // Ready to send connection data for some period of time
 #[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
-pub struct ConnectionSegmentData {
-    node: PublicKey,
+pub(crate) struct RelayConnDataSegment {
+    node: SocketAddr,
```

Comment:
nit: to make it more consistent maybe use the same name here (`server_address`)?